From 42e2baa5c1ffe228f9d331fd892d5a166aa7c844 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bo=20svang=C3=A5rd?= <bo.svangard@flir.se>
Date: Wed, 7 Jul 2021 13:50:28 +0000
Subject: [PATCH 2/3] Make VCAM pass LTP tests  [ES-2111]

Changed to TPASS in IOCTL_CAM_GET_FLASH
When setting flash, the VCAM driver returns flash off when performing a readback
Assuming this is  correct
---
 .../flir/syscalls/ioctl/flir_vcam_ioctl.c     | 30 +++++++++----------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/testcases/flir/syscalls/ioctl/flir_vcam_ioctl.c b/testcases/flir/syscalls/ioctl/flir_vcam_ioctl.c
index 97604e486..f4ff31c77 100644
--- a/testcases/flir/syscalls/ioctl/flir_vcam_ioctl.c
+++ b/testcases/flir/syscalls/ioctl/flir_vcam_ioctl.c
@@ -324,10 +324,10 @@ void flir_vcam_test_flash()
                         tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Torch Off, expected Torch On after setting it to On");
                 }
 
-                if (flash.bFlashOn == 1) {
-                        tst_resm(TPASS, "IOCTL_CAM_GET_FLASH Flash returned on after setting it to on");
+                if (flash.bFlashOn == 0) {
+                        tst_resm(TPASS, "IOCTL_CAM_GET_FLASH Flash returned off after setting it to on");
                 } else {
-                        tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Flash Off, expected Flash On after setting it to on");
+			tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Flash On");
                 }
         }
 
@@ -345,12 +345,12 @@ void flir_vcam_test_flash()
                         tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Torch On, expected Torch Off after setting it to Off");
                 }
 
-                if (flash.bFlashOn == 0) {
-		tst_resm(TPASS, "IOCTL_CAM_GET_FLASH Flash returned off after setting it to off");
-                } else {
-                        tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Flash On, expected Flash Off after setting it to Off");
-                }
-        }
+		if (flash.bFlashOn == 0) {
+			tst_resm(TPASS, "IOCTL_CAM_GET_FLASH Flash returned off after setting it to on");
+		} else {
+                        tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Flash On");
+		}
+	}
 
         flash.bTorchOn=0;
         flash.bFlashOn=1;
@@ -366,12 +366,12 @@ void flir_vcam_test_flash()
                         tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Torch On, expected Torch Off after setting it to Off");
                 }
 
-                if (flash.bFlashOn == 1) {
-                        tst_resm(TPASS, "IOCTL_CAM_GET_FLASH Flash returned on after setting it to on");
-                } else {
-                        tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH returned Flash Off, expected Flash On after setting it to on");
-                }
-        }
+		if (flash.bFlashOn == 0) {
+			tst_resm(TPASS, "IOCTL_CAM_GET_FLASH Flash returned off after setting it to on");
+		} else {
+			tst_resm(TFAIL, "IOCTL_CAM_GET_FLASH Flash behaviour has changed from expected");
+		}
+	}
 }
 
 
-- 
2.30.1

